tools/convert_hf_to_sequential.py

```diff
@@ -379,45 +396,72 @@ def convert(hf_model, ckpt_dir, output_dir):
         action="store_true",
         help="If set, will run a test to ensure the conversion was successful."
     )
-    parser.add_argument(
-        "--deduped",
-        action="store_true",
-        help="If set, will load the deduped version of the model."
-    )
     parser.add_argument(
         "--download-only",
         action="store_true",
         help="If set, script will only download the model and not convert it."
     )
 
+    parser.add_argument(
+        "--ckpt-tmp-dir",
+        default="/tmp/ckpt_tmp_dir",
+        help="Directory to store cached hugging face checkpoints. [WARNING: MUST BE VISIBLE TO ALL RANKS]"
+    )
+    parser.add_argument(
+        "--hf-model-name",
+        type=str,
+        help="Name of the hugging face model to download from EleutherAI/{hf-model-name}.}"
+    )
+
     parser.add_argument(
         "--cache-dir",
         default="/gpfs/alpine/csc499/proj-shared/hf_checkpoints",
         help="Directory to store cached hugging face checkpoints."
     )
     try:
-        if os.environ['WORLD_SIZE'] > 1:
+        if int(os.environ['WORLD_SIZE']) > 1:
             args = parser.parse_args(MULTI_GPU_ARGS.split(" "))
         else:
             args = parser.parse_args()
     except KeyError:
         args = parser.parse_args()
 
 
-    hf_model = GPTNeoXForCausalLM.from_pretrained(
-        f"EleutherAI/pythia-{args.size}-deduped" if args.deduped else f"EleutherAI/pythia-{args.size}" ,
-        revision=f"step{args.revision}",
-        cache_dir=os.path.join(args.cache_dir,f"pythia-{args.size}-deduped/step{args.revision}")
-    ).half()
+    tmp_cache_dir = get_non_existing_dir(args.ckpt_tmp_dir)
 
     if args.download_only:
+        hf_model = GPTNeoXForCausalLM.from_pretrained(
+            f"EleutherAI/{args.hf_model_name}",
+            revision=f"step{args.revision}",
+            cache_dir=os.path.join(args.cache_dir,f"{args.hf_model_name}/step{args.revision}")
+        ).half()
         exit(0)
+    else:
+        print("======================================================================")
+        print("Warning the following script will delete files withing {}".format(args.output_dir))
+        print("Warning the following script will delete this directory {}".format(tmp_cache_dir))
+        print("======================================================================")
+        # time.sleep(5)
 
-    neox_args = NeoXArgs.from_ymls(args.config)
+    
+    if int(os.environ.get('OMPI_COMM_WORLD_SIZE',1)) > 1:
+        neox_args = consume_neox_args2(args2)
+    else:
+        neox_args = NeoXArgs.from_ymls(args.config)
     neox_args.configure_distributed_args()
-    neox_args.build_tokenizer() 
+    neox_args.build_tokenizer()
+    neox_args.initialize_tensorboard_writer()
+
+
+    # setup logging and timers
+    init_wandb(neox_args=neox_args)
+    timers = Timers(
```

Comment:
What's the point of this timer?

Reply:
The point is to mimic the training code to a T as that has been thoroughly tested

Reply:
I understand the logic, but timing/logging is something we can safely ignore and I don't want wandb to be a dependency of this script.

Comment out lines 5, 23-26, and 457-460.

Reply:
Fair enough. I've commended those lines and also verified the script works on Summit.